add validation on url query params

Omarabdul3ziz opened the pull request:
- separate decoding into filters and limits
- return error if a passed param does not fit on a filter/limit

### Related Issues
- https://github.com/threefoldtech/tfgrid-sdk-go/issues/461

### Checklist

- [x] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstring

Omarabdul3ziz committed:
add validation on url query params:
- seperate decoding into filters and limits
- return error if a passed param does not fit on a filter/limit

Omarabdul3ziz committed:
Merge branch 'development' of https://github.com/threefoldtech/tfgrid-sdk-go into development_proxy_params_validation

xmonader reviewed (changes requested):
I don't understand the PR, please make sure to add comments and tests

Omarabdul3ziz committed:
add document and tests:
- add comments on the query parser part, documents what each step do and why it is needed.
- add tests covers the possible scenarios for the parse methods

xmonader commented on grid-proxy/internal/explorer/param_parser_test.go:
```diff
@@ -1,74 +1,72 @@
+	assert.Equal(t, expectedLimit, gotLimit)
+}
 
+func TestUnknownParam(t *testing.T) {
```
Add another test for a known and unknown param in the same query string

Omarabdul3ziz committed:
add mixed known and unknown params for testing param parser

xmonader commented on grid-proxy/internal/explorer/param_parser.go:
```diff
@@ -1,32 +1,93 @@
+	newParams := make(url.Values)
+	for key, val := range params {
+		for _, v := range val {
+			if v != string("") {
```
why `string("")` instead of `""`?

Omarabdul3ziz replied:
i think I was testing something and didn't clean it :D. removed now

Omarabdul3ziz committed:
remove unnecessary empty string casting
